### PR TITLE
[KYUUBI #884][FOLLOWUP] Fix catalog in KyuubiDatabaseMetaData#getTables

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

The behavior of `KyuubiDatabaseMetaData#getTables` changed by accident in #884[1], this PR restores the method's previous behavior and adds multi-catalog test case to verify it works as expected.

[1] [This](https://github.com/apache/incubator-kyuubi/commit/1a8b4ebaaeb8308018cab0c65497c74bc38f4a85#diff-c540737d1b293b909d07d50a05f67d35d274bc3163d723409135634a575e5e9dR68) makes KyuubiDatabaseMetaData#getTables method make no sense, just invokes the superclass.

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
> The behavior of KyuubiDatabaseMetaData#getTables changed by accident in #884

Can you detail the behavior change in the PR desc?

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/KyuubiDatabaseMetaData.java [hunk 1]
```diff
@@ -60,11 +60,10 @@ public ResultSet getTables(String catalog, String schemaPattern,
         if (tStatus.getStatusCode() != TStatusCode.SUCCESS_STATUS) {
             throw new HiveSQLException(tStatus);
         }
-        new HiveQueryResultSet.Builder(conn)
+        return new HiveQueryResultSet.Builder(conn)
                 .setClient(client)
                 .setSessionHandle(sessHandle)
                 .setStmtHandle(getTableResp.getOperationHandle())
                 .build();
-        return super.getTables(catalog, schemaPattern, tableNamePattern, types);
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/KyuubiDatabaseMetaData.java:68]: It makes the method make no sense, just invokes the superclass.

Comment (pan3793) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/KyuubiDatabaseMetaData.java:68]: @yaooqinn FYI

Comment (yaooqinn) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/KyuubiDatabaseMetaData.java:68]: Iet's update in the PR desc then

Comment (pan3793) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/KyuubiDatabaseMetaData.java:68]: Updated.

**Comment by pan3793** [comment]
cc @yaooqinn, could you please take a look again?

File: kyuubi-hive-jdbc/src/test/scala/org/apache/kyuubi/jdbc/KyuubiDriverSuite.scala [hunk 2]
```diff
@@ -19,33 +19,55 @@ package org.apache.kyuubi.jdbc
...
+      assert(resultSet1.getString(3) === "kyuubi_hive_jdbc")
+
+      val resultSet2 = metaData.getTables(catalog, "default", "%", null)
+      assert(resultSet2.next())
+      assert(resultSet2.getString(1) === catalog)
+      assert(resultSet2.getString(2) === "default")
+      assert(resultSet2.getString(3) === "hdp_cat_tbl")
     } finally {
-      statement.execute(s"DROP TABLE ${table}")
+      statement.execute(s"DROP TABLE $table1")
```
Comment (yaooqinn) [inline kyuubi-hive-jdbc/src/test/scala/org/apache/kyuubi/jdbc/KyuubiDriverSuite.scala:66]: nit: IF EXISTS

File: kyuubi-hive-jdbc/src/test/scala/org/apache/kyuubi/jdbc/KyuubiDriverSuite.scala [hunk 3]
```diff
@@ -19,33 +19,55 @@ package org.apache.kyuubi.jdbc
...
+      statement.execute(s"DROP TABLE $table2")
```
Comment (yaooqinn) [inline kyuubi-hive-jdbc/src/test/scala/org/apache/kyuubi/jdbc/KyuubiDriverSuite.scala:67]: ditto

File: kyuubi-hive-jdbc/src/test/scala/org/apache/kyuubi/jdbc/KyuubiDriverSuite.scala (hunk 2 above)
Comment (pan3793) [reply kyuubi-hive-jdbc/src/test/scala/org/apache/kyuubi/jdbc/KyuubiDriverSuite.scala]: Updated.

**Review by yaooqinn — APPROVED** [review]
LGTM

**Comment by pan3793** [comment]
Thanks for review, merging to master/1.3

**Pull request closed without merging**